elastic/kibana
# [Snapshot restore] - Hide system indices when creating a policy and restoring a snapshot

Addresses https://github.com/elastic/kibana/issues/120314

### Summary

With https://github.com/elastic/elasticsearch/pull/79670, we can no longer use the `indices` parameter of the ES create snapshot policy API or the ES restore snapshot API to back up or restore system indices in 8.0+ as these operations will result in errors.

This PR makes two main changes:
* Disallows manually selecting system_indices/ds when creating a policy
* Disallows manually selecting system_indices/ds when restoring a snapshot

**Steps to test:**
1. Run elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/es-backups` and also run kibana
2. Navigate to console and run the following command to create a repository:
```
.PUT /_snapshot/my_backup
{
  "type": "fs",
  "settings": {
    "location": "/tmp/es-backups",
    "chunk_size": "10mb"
  }
}
```
3. Go to the homepage and add sample data so that we have some indices to play with
4. Go to snapshot and restore app and create a new policy
5. Verify that when disabling the toggle from the `Data streams and indices` in step 2 you don't see any system indices
6. Go ahead and create the policy keeping global state or by  keeping the before mentioned toggle **on** so that we end up with system indices in the snapshot and run the policy
7. Go to the snapshots tab and verify that in the details flyout there are system indices
8. Try to restore the snapshot and make sure that when disabling the toggle from the `Data streams and indices` in step 2 you don't see any system_indices/ds


Note: Review will be easier if you set hide whitespace changes
